src/changelog/utils.py

```diff
@@ -36,30 +38,30 @@ def get_changelog_data(self):
             data = changelog.readlines()
```

Comment:
Seeing the `List[str]` where I was expecting a `Iterable[str]` and seeing it be correct probably is worth filing an enhancement where we dont need to readout and writeout the entire file for every change. Overall changelogs are not _large_ documents, but being efficient is probably worth looking into and instead consuming an iterator until conditions are met, since very often we only need the first few entires in the file.

Reply:
Very good point. I've logged it as https://github.com/mc706/changelog-cli/issues/26. Should be a pretty simple fix for another PR.